bowtie/_core.py

```diff
@@ -625,6 +643,144 @@ async def start(
 
         yield cls(harness=harness, id=id, info=info, reporter=reporter)
 
+    @classmethod
+    async def download_versions_of(cls, id: ConnectableId) -> Set[str] | None:
```

Comment:
Try to avoid importing and using rich in this file -- rich is about presentation, and specifically presentation to the TUI.

Elsewhere we've tried (although perhaps not fully succeeded) at returning data structures, and then converting to rich only when needed.

So here it seems like maybe this should simply return a deserialized thing from retrieving the HTTP Response. Possibly as a new type.